tls_codec/src/lib.rs

```diff
@@ -198,6 +236,41 @@ pub trait Deserialize: Size {
     }
 }
 
+#[cfg(hax)]
+pub trait DeserializeExact: Deserialize {
+    /// This function deserializes the provided `bytes` and returns the populated
+    /// struct. All bytes must be consumed.
+    ///
+    /// Returns an error if not all bytes are read from the input, or if an error
+    /// occurs during deserialization.
+    #[cfg(feature = "std")]
+    fn tls_deserialize_exact(bytes: impl AsRef<[u8]>) -> Result<Self, Error>
+    where
+        Self: Sized;
+}
+
+#[cfg(hax)]
+impl<T: Deserialize> DeserializeExact for T {
+    /// This function deserializes the provided `bytes` and returns the populated
+    /// struct. All bytes must be consumed.
+    ///
+    /// Returns an error if not all bytes are read from the input, or if an error
+    /// occurs during deserialization.
+    #[cfg(feature = "std")]
+    fn tls_deserialize_exact(bytes: impl AsRef<[u8]>) -> Result<Self, Error>
+    where
+        Self: Sized,
+    {
+        let mut bytes = bytes.as_ref();
```

Comment:
We can move out the actual code to a standalone function to avoid duplicating code. Then the extraction would also have the actual code that's being run.